doc/components.md

```diff
@@ -32,13 +32,13 @@ Every `Component` has a few other methods that you can optionally implement, tha
 
 The `resize` method is called whenever the screen is resized, and in the beginning once when the component is added via the `add` method. You need to apply here any changes to the x, y, width and height of your component, or any other changes, due to the screen resizing. You can start these variables here, as the sprite won't be rendered until everything is set.
 
-The `destroy` method can be implemented to return true and warn the `BaseGame` that your object is marked for destruction, and it will be remove after the current update loop. It will then no longer be rendered or updated.
+The `shouldRemove` method can be implemented to return true and `BaseGame` will remove it before the next update loop. It will then no longer be rendered or updated. Note that `game.remove(Component c)` can also be used to remove components.
 
 The `isHUD` method can be implemented to return true (default false) to make the `BaseGame` ignore the `camera` for this element.
 
 The `onMount` method can be overridden to run initialization code for the component. When this method is called, BaseGame ensures that all the mixins which would change this component behaviour are already resolved.
 
-The `onDestroy` method can be overridden to run code before the component is removed from the game.
+The `onRemove` method can be overridden to run code before the component is removed from the game.
```

Comment:
I think this also fixes a bug where onRemove was called twice on v1. I pointed this out on telegram. I think we can make it explicit that is desired here? maybe can even add a test for that

Reply:
Good idea, more tests to the people!

Reply:
Done! Did quite a lot of other changes too, so would be good if you could re-review.